tests/Unit/APQTest.php

```diff
@@ -0,0 +1,359 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rebing\GraphQL\Tests\Unit;
+
+use Illuminate\Http\UploadedFile;
+use Rebing\GraphQL\Support\UploadType;
+use Rebing\GraphQL\Tests\TestCase;
+use Rebing\GraphQL\Tests\Unit\UploadTests\UploadMultipleFilesMutation;
+use Rebing\GraphQL\Tests\Unit\UploadTests\UploadSingleFileMutation;
+
+class APQTest extends TestCase
+{
+    protected function getEnvironmentSetUp($app)
+    {
+        parent::getEnvironmentSetUp($app);
+
+        $graphqlSchemasDefault = $app['config']->get('graphql.schemas.default');
+        $app['config']->set('graphql.schemas.default', array_merge_recursive($graphqlSchemasDefault, [
+            'mutation' => [
+                UploadMultipleFilesMutation::class,
+                UploadSingleFileMutation::class,
+            ],
+        ]));
+
+        $graphqlTypes = $app['config']->get('graphql.types');
+        $app['config']->set('graphql.types', array_merge_recursive($graphqlTypes, [
+            UploadType::class
+        ]));
+    }
+
+    /**
+     * Test persisted query not supported.
+     */
+    public function testPersistedQueryNotSupported(): void
+    {
+        config(['graphql.apq.enable' => false]);
+
+        $response = $this->call('GET', '/graphql', [
+            'extensions' => [
+                'persistedQuery' => [
+                    'version' => 1,
+                    'sha256Hash' => hash('sha256', trim($this->queries['examples'])),
```

Comment:
This because somewhere I have a feeling that the input query it comes trimmed, do you have a better solution in mind?

Reply:
No 😬

I wanted to understand the reason, because when I removed the `trim()` in the tests, it failed. But it's not clear what the reason for this is…

Still would like to "know" where it happens but haven't yet figured it out.

Reply:
The first feeling that comes to my mind is that it comes from `$this->call` (\Illuminate\Foundation\Testing\Concerns\MakesHttpRequests)

Reply:
Found it!

I don't use this personally so I didn't think about it, but testbench-core by default includes the `TrimStrings` middleware => https://github.com/orchestral/testbench-core/blob/6699f72131e73b889ee25747aefc90f7401f84b8/src/Http/Kernel.php#L18 which AFAIK is common to be used in Laravel.

So whether triming is necessary or not depends on whether the middleware is enabled or not.

I've added a note to the README about his, could trip people having it enabled (deliberately or not).